### PR TITLE
T27152 Hack mode settings

### PR DESCRIPTION
**Pull request by joaquimrocha**
https://phabricator.endlessm.com/T27152

### CODE REVIEW

File: js/ui/codeView.js [hunk 1]
```diff
@@ -1045,6 +1051,11 @@ var CodingSession = GObject.registerClass({
         let primaryMonitor = Main.layoutManager.primaryMonitor;
         let inFullscreen = primaryMonitor && primaryMonitor.inFullscreen;
 
+        if (!global.settings.get_boolean('hack-mode-enabled')) {
+            this._button.hide();
+            return;
+        }
+
```
Comment (cfoch) [inline js/ui/codeView.js]: Maybe move this code after 
```
        if (!focusedWindow)
            return;
```

File: data/org.gnome.shell.gschema.xml.in [hunk 2]
```diff
@@ -126,16 +126,6 @@
         Whether the application grid displays the App Center icon.
       </description>
     </key>
-    <key name="enable-code-view" type="b">
```
Comment (cfoch) [inline data/org.gnome.shell.gschema.xml.in]: I think you should squash it with the commit "data: Add hack-mode-enabled setting"

**Comment by cfoch** [comment]
I left some comments but are very trivial. In general it is okay. So feel free to do the changes or not (maybe if there is something more important that you are working on right now).

**Comment by danigm** [comment]
This is merged in the `hack2` branch.

**Comment by danigm** [comment]
Closing, this is master now.

**Pull request closed without merging**